File: plusmall-operator-interface/src/main/java/com/plusmall/operator/ItemCatService.java
package com.plusmall.operator;

import com.plusmall.commons.PageResult;
import com.plusmall.model.TbItemCat;

import java.util.List;

/**
 * @Description:
 */
public interface ItemCatService {
	public PageResult searchByParentId(int pageNum,int pageSize,Long parentId);
	public void delete(Long[] ids) throws NullPointerException;
	public void add(TbItemCat itemCat) throws NullPointerException;
	public TbItemCat findOne(Long id);
	public void update(TbItemCat itemCat) throws NullPointerException;
	public List<TbItemCat> findAll();
}

File: plusmall-business-web/src/main/webapp/js/service/itemCatService.js
app.service('itemCatService',function ($http) {

    //通过parentId查找商品分类
    this.searchByParentId=function (parentId) {
        return $http.get('../itemcat/searchByParentId.do?parentId='+parentId);
    }

    //通过id查找单个分类数据
    this.findOne=function (id) {
        return $http.get('../itemcat/findOne.do?id='+id);
    }

    //查找商品分类表中的全部信息
    this.findAll=function () {
        return $http.get('../itemcat/findAll.do');
    }
})